File: scripts/get-members.ts
require('dotenv').config({
    path: __dirname + '/.env.local',
})

import config from '../config'
import { Octokit } from '@octokit/rest'
import fs from 'fs'



console.log('access_token', process.env.access_token)
const octokit = new Octokit({
    auth: process.env.access_token,
})

async function fetch(){
    const ret = await Promise.all(
        config.members.map((member) =>
            octokit.users.getByUsername({ username: member })
        )
    )
    return ret.map((user, index) => ({
        url: user.data.html_url,
        id: user.data.id,
        name: user.data.name ?? config.members[index],
        avatar: user.data.avatar_url,
        company: user.data.company,
        blog: user.data.blog,
        email: user.data.email,
        bio: user.data.bio,
        location: user.data.location
    }))
}

async function main() {
    const members = await fetch()
    console.log(members)
    fs.writeFileSync(
        __dirname + '/../generated/members.json',
        JSON.stringify(members, undefined, 4)
    )
    console.log('done')
}
main()

File: scripts/get_members.py
# coding=utf-8

from github import Github
import sys
import json

members = [
    'stkevintan',
    'gbzhu',
    'xuliker',
    '0xHJK',
    'Soulghost',
    'SRaddict',
    'Superlee2013',
]


def get_token():
    flag = False
    for s in sys.argv:
        if flag:
            return s
        if s.strip() == '--token':
            flag = True
    return None


def main():
    g = Github(get_token())

    def get_user(name: str):
        u = g.get_user(name)
        return {
            'url': u.html_url,
            'id': u.id,
            'name': u.name if u.name else name,
            'avatar': u.avatar_url,
            'company': u.company,
            'blog': u.blog,
            'email': u.email,
            'bio': u.bio,
            'location': u.location
        }

    users = map(get_user, members)
    with open('./generated/members.json', 'w') as f:
        json.dump(list(users), f, ensure_ascii=False)


if __name__ == "__main__":
    main()
